src/lib.rs

```diff
@@ -314,38 +314,43 @@ pub struct ApiError {
 }
 
 impl Mastodon {
-    fn from_registration<I>(base: I,
-                         client_id: I,
-                         client_secret: I,
-                         redirect: I,
-                         token: I,
-                         client: Client)
-        -> Self
-        where I: Into<Cow<'static, str>>
-        {
-            let data = Data {
-                base: base.into(),
-                client_id: client_id.into(),
-                client_secret: client_secret.into(),
-                redirect: redirect.into(),
-                token: token.into(),
-
-            };
-
-            let mut headers = Headers::new();
-            headers.set(Authorization(Bearer { token: (*data.token).to_owned() }));
-
-            Mastodon {
-                client: client,
-                headers: headers,
-                data: data,
-            }
+    fn from_registration<I>(
+        base: I,
+        client_id: I,
+        client_secret: I,
+        redirect: I,
+        token: I,
+        client: Client,
+    ) -> Self
+    where
+        I: Into<Cow<'static, str>>,
```

Comment:
If `-> Self` was on it's own line and the `where` clause was adjoined with the next line to make `where I: Into<Cow<'static, str>>,` I'd accept this.